opt-tools/docker/docker-compose.yml

```diff
@@ -0,0 +1,57 @@
+version: '3'
```

Comment:
I tried to bring up your application but the next error was occurred:
`networks.docker-practice-net value Additional properties are not allowed ('name' was unexpected)`

Network names were added in 3.5 version of Docker-compose: https://docs.docker.com/compose/compose-file/compose-versioning/#version-35
So, please let's set docker-compose version explicitly.

Reply:
Why does everything work well for me then?

Reply:
But I'll fix it of course.

Reply:
About: "This exercise requires to change application.properties within the spring.datasource.url at least, or redefine this setting via environment variables somehow, but I cannot found such change. Could you please explain how backend should reach the database? (the current default application.properties has a localhost as the database target for Java app)."
I understand that need to replace these lines
spring.datasource.url=jdbc:postgresql://localhost:5432/school
spring.redis.host=localhost
with these
spring.datasource.url=jdbc:postgresql://db:5432/school
spring.redis.host=redis

Reply:
>Why does everything work well for me then?
Probably it is related to host platform and how the docker-compose massages with its version definition on docker-compose file. And as I mentioned early - let's be a little bit more precise within the: versions of packages, images, and other stuff.

>with these
>spring.datasource.url=jdbc:postgresql://db:5432/school
>spring.redis.host=redis
Yep, you understood it correctly!